test/jdk/sun/net/www/http/KeepAliveCache/KeepAliveProperty.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @library /test/lib
+ * @bug 8278067
+ * @run main/othervm -Dhttp.keepAlive.time.server=30 KeepAliveProperty long
+ * @run main/othervm -Dhttp.keepAlive.time.server=1 KeepAliveProperty short
+ * @run main/othervm -ea -Dhttp.keepAlive.time.server=0 KeepAliveProperty short
+ */
+
+import java.net.*;
+import java.io.*;
+import java.nio.charset.*;
+import java.util.logging.*;
+import jdk.test.lib.net.URIBuilder;
+import static java.net.Proxy.NO_PROXY;
+
+public class KeepAliveProperty {
+
+    static volatile boolean pass = false;
+
+    static class Server extends Thread {
+        ServerSocket server;
+        Server (ServerSocket server) {
+            super ();
+            this.server = server;
+        }
+        void readAll (Socket s) throws IOException {
+            byte[] buf = new byte [128];
+            int c;
+            String request = "";
+            InputStream is = s.getInputStream ();
+            while ((c=is.read(buf)) > 0) {
+                request += new String(buf, 0, c, StandardCharsets.US_ASCII);
+                if (request.contains("\r\n\r\n")) {
+                    return;
+                }
+            }
+        }
+
+
+        Socket s = null;
+        String BODY;
+        String CLEN;
+        PrintStream out;
+
+        public void run() {
+            try {
+                s = server.accept();
+                readAll(s);
+
+                BODY = "Hello world";
+                CLEN = "Content-Length: " + BODY.length() + "\r\n";
+                out = new PrintStream(new BufferedOutputStream(s.getOutputStream() ));
+
+                /* send the header */
+                out.print("HTTP/1.1 200 OK\r\n");
+                out.print("Content-Type: text/plain; charset=iso-8859-1\r\n");
+                out.print(CLEN);
+                out.print("\r\n");
+                out.print(BODY);
+                out.flush();
+            } catch (Exception e) {
+                pass = false;
+                try {
+                    if (s != null)
+                        s.close();
+                    server.close();
+                } catch (IOException unused) {}
+                return;
+            }
+
+            // second request may legitimately fail
+
+            try {
+                // wait for second request.
+                readAll(s);
+
+                BODY = "Goodbye world";
+                CLEN = "Content-Length: " + BODY.length() + "\r\n";
+
+                /* send the header */
+                out.print("HTTP/1.1 200 OK\r\n");
+                out.print("Content-Type: text/plain; charset=iso-8859-1\r\n");
+                out.print(CLEN);
+                out.print("\r\n");
+                out.print(BODY);
+                out.flush();
+                pass = true;
```

Comment:
should that be: pass = ! expectClose?

Reply:
Good catch. That was masking another problem in the test where it wasn't detecting the socket close properly.